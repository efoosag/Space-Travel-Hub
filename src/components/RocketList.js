import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Rocket from './Rocket';
import { getRockets } from '../redux/rockets';

const RocketList = () => {
  const rockets = useSelector((state) => state.entities.rockets);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getRockets());
  });
  return (
    <section className="section">
      <ul className="list">
        {rockets.map((rocket) => (
          <li key={rocket.id}>
            <Rocket rocket={rocket} />
          </li>
        ))}
      </ul>
    </section>
  );
};

export default RocketList;
