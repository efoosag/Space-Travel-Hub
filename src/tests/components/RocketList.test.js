import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import RocketList from '../../components/RocketList';

describe('Testing RocketList component', () => {
  it('Testing with snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <RocketList />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
