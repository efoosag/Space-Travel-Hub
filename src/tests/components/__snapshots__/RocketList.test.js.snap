// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing RocketList component Testing with snapshot 1`] = `
<section
  className="section"
>
  <ul
    className="list"
  />
</section>
`;
